lib/Echidna/Solidity.hs

```diff
@@ -175,17 +191,22 @@ loadSpecified name cs = do
     unless q . putStrLn $ "Analyzing contract: " <> c ^. contractName . unpacked
 
   -- Local variables
-  (SolConf ca d ads bala balc pref _ _ libs _ fp ma ch) <- view hasLens
+  (SolConf ca d ads bala balc pref _ _ libs _ fp ma ch fs) <- view hasLens
 
   -- generate the complete abi mapping
-  let abiOf :: SolcContract -> NE.NonEmpty SolSignature
-      abiOf cc = fallback NE.:| filter (not . isPrefixOf pref . fst) (elems (cc ^. abiMap) <&> \m -> (m ^. methodName, m ^.. methodInputs . traverse . _2))
-      abiMapping = if ma then M.fromList $ cs <&> \cc -> (cc ^. runtimeCode . to stripBytecodeMetadata, abiOf cc) else M.singleton (c ^. runtimeCode . to stripBytecodeMetadata) (abiOf c)
-      bc = c ^. creationCode
+  let bc = c ^. creationCode
       abi = liftM2 (,) (view methodName) (fmap snd . view methodInputs) <$> toList (c ^. abiMap)
       con = view constructorInputs c
       (tests, funs) = partition (isPrefixOf pref . fst) abi
-
+      
+
+  -- Filter ABI according to the config options
+  fabiOfc <- filterMethods fs $ abiOf pref c
```

Comment:
is this the correct place to do this? how does this affect contracts created afterwards and multi-abi interaction?

Reply:
Filtering methods is not supported in multi-abi mode. We can extend the support to that later, if necessary.